S/StableHLO/build_tarballs.jl

```diff
@@ -0,0 +1,111 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+const YGGDRASIL_DIR = "../.."
+include(joinpath(YGGDRASIL_DIR, "fancy_toys.jl"))
+include(joinpath(YGGDRASIL_DIR, "platforms", "llvm.jl"))
+
+name = "StableHLO"
+version = v"0.14.6"
+
+sources = [
+    GitSource("https://github.com/openxla/stablehlo.git", "8816d0581d9a5fb7d212affef858e991a349ad6b"),
+]
+
+llvm_versions = [v"17.0.6+0"]
+
+script = raw"""
+cd $WORKSPACE/srcdir/stablehlo
+
+# need to run mlir-tblgen and mlir-pdll on the host
+rm ${prefix}/bin/mlir-tblgen ${prefix}/bin/mlir-pdll
+ln -s ${host_prefix}/bin/mlir-tblgen ${prefix}/bin/mlir-tblgen
+ln -s ${host_prefix}/bin/mlir-pdll ${prefix}/bin/mlir-pdll
+
+CMAKE_FLAGS=()
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING:BOOL=ON)
+
+CMAKE_FLAGS+=(-DLLVM_DIR=${prefix}/lib/cmake/llvm)
+CMAKE_FLAGS+=(-DLLVM_ENABLE_ASSERTIONS=OFF)
+CMAKE_FLAGS+=(-DLLVM_LINK_LLVM_DYLIB=ON)
+CMAKE_FLAGS+=(-DLLVM_ENABLE_LLD="OFF")
+
+CMAKE_FLAGS+=(-DMLIR_DIR=${prefix}/lib/cmake/mlir)
+
+# building shared libraries crashes on linking, so build static libraries and link them manually
+CMAKE_FLAGS+=(-DBUILD_SHARED_LIBS=OFF)
+
+cmake -B build -S . -GNinja ${CMAKE_FLAGS[@]}
+ninja -C build -j 1 install
+
+# NOTE no idea why they are not installed by default
+install build/bin/stablehlo-opt build/bin/stablehlo-translate build/bin/stablehlo-lsp-server ${prefix}/bin
+
+# build shared library from static libraries
+cd build/lib
+for i in *.a; do
+    echo $i
+    ar x $i
+done
+c++ -shared -o libStablehlo.${dlext} -lLLVM -lMLIR *.cpp.o
+install libStablehlo.${dlext} ${prefix}/lib
```

Comment:
Verbose command and more specific env var
```suggestion
install -v libStablehlo.${dlext} ${libdir}
```